src/transformers/models/t5/tokenization_t5.py

```diff
@@ -349,15 +366,10 @@ def _tokenize(self, text, **kwargs):
         the extra `SPIECE_UNDERLINE` prepended.
         """
         if not self.legacy:
-            is_first = text.startswith(SPIECE_UNDERLINE)
-            if is_first:
-                text = text[1:]
-
-        tokens = self.sp_model.encode(text, out_type=str)
-
-        if not self.legacy and not is_first and not text.startswith(" ") and tokens[0].startswith(SPIECE_UNDERLINE):
-            tokens = ([tokens[0][1:]] if len(tokens[0]) > 1 else []) + tokens[1:]
-        return tokens
+            text = self.unk_token + text
+            tokens = self.sp_model.encode(text, out_type=str)
+            return tokens[self.unk_token_length :]
```

Comment:
That's the Hack: 
- all spm models have a tokenizer. Whether or not it is in the sentencepiece vocab does not matter. 
- we need to do this because since `add_dummy_prefix = False` the sentencpiece model always ALWAYS strips any `SPIECE_UNDERLINE`. So `sp_model.encode(SPIECE_UNDERLINE + "Hello" , out_type=str)` will give [`Hel`,`llo`]  instead of [`_Hel`, `llo`]. 
- previously, we removed added extra space. This is okay, but fails for words that should be split like `inform`. What happened before was that we would tokenize as `_inform` then remove `_` and we have `inform`. But, the actual tokenization of `inform` is `in`,`form` and `inform` is not part of the vocab!